Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.ps1

```diff
@@ -1942,18 +1942,64 @@ function Remove-EmptyItems {
 
     $newDict = @{}
 
-    foreach ($property in $inputObject.PSObject.Properties) {
-        $value = $property.Value
+    if ($inputObject -is [hashtable]) {
+        # Handle hashtable input
+        foreach ($key in $inputObject.Keys) {
+            $value = $inputObject[$key]
+            $Demisto.Debug("Looking at hashtable key '$key' with value: '$value'")
+            
+            # Skip if value is null
+            if ($null -eq $value) {
+                $Demisto.Debug("Skipping null value for key '$key'")
+                continue
+            }
+            
+            # Skip if value is a string and is empty or whitespace
+            if ($value -is [string] -and [string]::IsNullOrWhiteSpace($value)) {
+                $Demisto.Debug("Skipping empty string for key '$key'")
+                continue
+            }
+            
+            # Skip if value is an empty collection
+            if ($value -is [System.Collections.IEnumerable] -and -not ($value -is [string]) -and
+                (($value | Measure-Object).Count -eq 0 -or $value.Count -eq 0)) {
+                $Demisto.Debug("Skipping empty collection for key '$key'")
+                continue
+            }
+            
+            # If we got here, the value is not empty, so add it to the new dictionary
+            $newDict[$key] = $value
+            $Demisto.Debug("Added key '$key' with value: '$value'")
+        }
+    } else {
+        # Handle PSObject input
+        foreach ($property in $inputObject.PSObject.Properties) {
+            $value = $property.Value
+            $propertyName = $property.Name
+            $Demisto.Debug("Looking at property '$propertyName' with value: '$value'")
+
+            # Skip if value is null
+            if ($null -eq $value) {
```

Comment:
Seems like a bit of duplicate code here. 
I think it will be better to add a helper function to check for null/empty values, and then use it in the loops
something like:
```
function Test-IsEmptyValue {
    param ($Value)

    if ($null -eq $Value) { return $true }

    if ($Value -is [string]) {
        return [string]::IsNullOrWhiteSpace($Value)
    }

    if ($Value -is [System.Collections.ICollection]) {
        return $Value.Count -eq 0
    }

    return $false
}

```